src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -51,12 +53,84 @@ public sealed class NpgsqlRawCopyStream : Stream, ICancelable
 
         #region Constructor
 
-        internal NpgsqlRawCopyStream(NpgsqlConnector connector, string copyCommand)
+        private async Task SendQueryAsync(NpgsqlConnector connector, NpgsqlStatement statement, bool async)
+        {
+            await connector.ParseMessage
+                .Populate(statement.SQL, statement.StatementName, statement.InputParameters, connector.TypeMapper)
+                .Write(_writeBuf, async);
+            var bind = connector.BindMessage;
+            bind.Populate(statement.InputParameters, "", statement.StatementName);
+            await connector.BindMessage.Write(_writeBuf, async);
+            await connector.DescribeMessage
+                .Populate(StatementOrPortal.Portal)
+                .Write(_writeBuf, async);
+            await ExecuteMessage.DefaultExecute.Write(_writeBuf, async);
+            await SyncMessage.Instance.Write(_writeBuf, async);
+            await _writeBuf.Flush(async);
+        }
+
+        private NpgsqlStatement ParseRawQuery(string copyCommand, NpgsqlParameterCollection parameters)
+        {
+            var statements = new List<NpgsqlStatement>();
+            _connector.SqlParser.ParseRawQuery(copyCommand, _connector.UseConformantStrings, parameters, statements, deriveParameters: false);
+
+            if (statements.Count < 0)
+            {
+                throw new ArgumentException("TODO - better message");
+            }
+
+            var copyStatement = statements[0];
+            // TODO: Should we do this?
+            //copyStatement.StatementType = StatementType.Copy;
+            return copyStatement;
+        }
+
+        void ValidateParameters(NpgsqlParameterCollection parameters)
```

Comment:
Again, it makes sense for most of the code to live in NpgsqlCommand, this way methods like `ValidateParameters()` get reused.